main/main.c

```diff
@@ -1752,6 +1752,11 @@ void php_request_shutdown(void *dummy)
 		php_call_shutdown_functions();
 	}
 
+	/* 1.5. Call any open end handlers that are still open after a zend_bailout */
+	if (ZEND_OBSERVER_ENABLED) {
+		zend_observer_fcall_end_all();
+	}
```

Comment:
Shouldn't this be before shutdown functions are called?

Reply:
You're right. The original idea was to include the shutdown handler time before closing out the open observed stack, but that doesn't make sense in this context. Thanks for pointing that out. I'll move it up.